provisioning/device/test/_x509_registration_test.js

```diff
@@ -26,25 +26,28 @@ describe('X509Registration', function () {
   describe('#register', function () {
 
     /* Tests_SRS_NODE_DPS_X509_REGISTRATION_18_001: [ `register` shall call `getCertificate` on the security object to acquire the X509 certificate. ] */
-    /* Tests_SRS_NODE_DPS_X509_REGISTRATION_18_002: [ `register` shall call `registerX509` on the transport object and call it's callback with the result of the transport operation. ] */
+    /* Tests_SRS_NODE_DPS_X509_REGISTRATION_18_002: [ `register` shall call `register` on the pollingStateMachine and call `callback` with the result. ] */
+    /* Tests_SRS_NODE_DPS_X509_REGISTRATION_18_004: [ `register` shall pass the certificate into the `setAuthentication` method on the transport ] */
     it ('gets an x509 cert and calls registerX509', function(callback) {
       var transport = {
-        registerX509: sinon.spy(function(request, auth, callback) { callback(null, fakeResponse); }),
-        getRegistrationId: function() { return fakeRegistrationId; }
+        setAuthentication: sinon.spy(),
       };
       var security = {
         getCertificate: sinon.spy(function(callback) { callback(null, fakeX509Cert); }),
         getRegistrationId: function() { return fakeRegistrationId; }
       };
       var clientObj = new X509Registration(fakeProvisioningHost, fakeIdScope, transport, security);
+      clientObj._pollingStateMachine.register = sinon.spy(function(request, callback) { callback(null, { registrationState: fakeResponse }); });
```

Comment:
>sinon.spy [](start = 48, length = 9)

in other tests across the SDK we use:
sinon.stub().callsFake(function () {})
and if it's just to call a callback with a default result:
sinon.stub().callsArgWith(<callbackArgIndex>, <error>, <result>)